### PR TITLE
sst_dump to reduce number of file reads

### PR DESCRIPTION
**Pull request by siying**
Summary: sst_dump can issue many file reads from the file system. This doesn't work well with file systems without a OS cache, especially remote file systems. In order to mitigate this problem, several improvements are done:
1. --readahead_size is added, so that users can specify readahead size when scanning the data.
2. Force a 512KB tail readahead, which prevents three I/Os for footer, meta index and property blocks and hopefully index and filter blocks too.
3. Consoldiate SSTDump's I/Os before opening the file for read. Use the same file prefetch buffer.

Test Plan: Add a test that covers this new feature.

### CODE REVIEW

File: tools/sst_dump_tool.cc [hunk 1]
```diff
@@ -318,8 +337,15 @@ Status SstFileDumper::SetTableOptionsByMagicNumber(
   assert(table_properties_);
   if (table_magic_number == kBlockBasedTableMagicNumber ||
       table_magic_number == kLegacyBlockBasedTableMagicNumber) {
-    options_.table_factory = std::make_shared<BlockBasedTableFactory>();
+    BlockBasedTableFactory* bbtf = new BlockBasedTableFactory();
+    // To force tail prefetching, we directly tail tail prefetching to
+    // read 512KB blocks.
+    bbtf->tail_prefetch_stats()->RecordEffectiveSize(512 * 1024);
+    bbtf->tail_prefetch_stats()->RecordEffectiveSize(512 * 1024);
```
Comment (pdillinger) [inline tools/sst_dump_tool.cc:344]: Twice?

File: tools/sst_dump_tool.cc [hunk 2]
```diff
@@ -318,8 +337,15 @@ Status SstFileDumper::SetTableOptionsByMagicNumber(
   assert(table_properties_);
   if (table_magic_number == kBlockBasedTableMagicNumber ||
       table_magic_number == kLegacyBlockBasedTableMagicNumber) {
-    options_.table_factory = std::make_shared<BlockBasedTableFactory>();
+    BlockBasedTableFactory* bbtf = new BlockBasedTableFactory();
+    // To force tail prefetching, we directly tail tail prefetching to
```
Comment (pdillinger) [inline tools/sst_dump_tool.cc:341]: tell tail prefetching

**Comment by pdillinger** [comment]
And of course fixing Windows and MacOS build errors

File: tools/sst_dump_tool.cc (hunk 1 above)
Comment (siying) [reply tools/sst_dump_tool.cc:344]: Yes. The stats only start to kick in after two.

**Pull request closed without merging**